autogpt_platform/frontend/src/components/DataTable.tsx

```diff
@@ -34,6 +83,22 @@ export default function DataTable({
     });
   };
 
+  const renderCellContent = (value: any) => {
+    if (typeof value === "string") {
+      if (isValidVideoUrl(value)) {
+        return <VideoRenderer videoUrl={value} />;
+      } else if (isValidImageUrl(value)) {
+        return <ImageRenderer imageUrl={value} />;
+      }
+    }
+
+    const text =
+      typeof value === "object" ? JSON.stringify(value) : String(value);
+    return truncateLongData && text.length > maxChars
+      ? text.slice(0, maxChars) + "..."
+      : text;
+  };
+
```

Comment:
extract this to a general content renderer so we can just 

```tsx
<Render content={content}/>
```

Reply:
Added ContentRenderer